src/coreclr/src/gc/gc.cpp

```diff
@@ -16802,8 +16749,10 @@ void gc_heap::gc1()
 
         if (n != max_generation)
         {
-            int gen_num_for_data = ((n < (max_generation - 1)) ? (n + 1) : (max_generation + 1));
-            for (int gen_number = (n + 1); gen_number <= gen_num_for_data; gen_number++)
+            // for gen0, update data for gen1
```

Comment:
I would not change this functionally - there's a reason why we were doing (n < (max_gen - 1)) because we want to expression if the gen is less than (the oldest gen - 1), if we ever want to have more than 2 ephemeral generations you just changed the meaning here.